### PR TITLE
Connect real api

### PR DESCRIPTION
**Pull request by Fewwy**
Connecting real API to the UI

Removed mocked data
Added a conditional rendering to hide objects table and button when the objects are absent for the recommendation.
Passed more data to the Expanded Rules Details to render the data using template processor
Conditional rendering to hide the Note part of the Expanded Rules Details component. It is dependent on the existence of a namespace Name.

### CODE REVIEW

File: src/Components/ExpandedRulesDetails.js/ExpandedRulesDetails.js [hunk 1]
```diff
@@ -20,11 +20,20 @@ import PropTypes from 'prop-types';
 import TemplateProcessor from '@redhat-cloud-services/frontend-components-advisor-components/TemplateProcessor/TemplateProcessor';
 import ObjectsModal from '../ObjectsModal/ObjectsModal';
 import { ObjectsTableColumns } from '../../AppConstants';
+import _ from 'lodash';
```
Comment (mkholjuraev) [inline src/Components/ExpandedRulesDetails.js/ExpandedRulesDetails.js:23]:
```suggestion
import isEmpty from 'lodash/isEmpty';
```

This helps to load only what is needed.

File: src/Components/WorkloadRules/WorkloadRules.js [hunk 2]
```diff
@@ -143,6 +143,7 @@ const WorkloadRules = ({ workload }) => {
   };
 
   const buildFilteredRows = (allRows, filters) => {
+    console.log(allRows, 'allrows');
```
Comment (mkholjuraev) [inline src/Components/WorkloadRules/WorkloadRules.js:146]: We can remove this, I believe

**Review by mkholjuraev — COMMENTED** [review]
I have run the app, I could see that now real API is used and there are no issues related to it. Let's fix the tests and I will have another quick look and probably we can proceed.

File: src/Components/WorkloadRules/WorkloadRules.js (hunk 2 above)
Comment (Fewwy) [reply src/Components/WorkloadRules/WorkloadRules.js:146]: Yes, forgot to delete that

File: src/Components/ExpandedRulesDetails.js/ExpandedRulesDetails.js [hunk 3]
```diff
@@ -20,11 +20,20 @@ import PropTypes from 'prop-types';
...
-const ExpandedRulesDetails = ({ more_info, resolution, objects }) => {
+const ExpandedRulesDetails = ({
+  more_info,
+  resolution,
+  objects,
+  namespaceName,
+  reason,
+  extra_data,
+}) => {
+  const objectsArePresent = !isEmpty(objects);
```
Comment (AsToNlele) [inline src/Components/ExpandedRulesDetails.js/ExpandedRulesDetails.js:36]: Just a small note, maybe we could make the bundle a bit smaller by removing the lodash function. I believe it always an array so we could check it like this
```suggestion
  const objectsArePresent = Array.isArray(objects) && objects.length > 0;
```

**Review by AsToNlele — APPROVED** [review]
Works as expected, good job! :smile: 
Left a small note

**Pull request merged**